valhalla/sif/dynamiccost.h

```diff
@@ -1001,7 +1014,7 @@ class DynamicCost {
     // Cases with both time and penalty: country crossing, ferry, rail_ferry, gate, toll booth
     sif::Cost c;
     c += country_crossing_cost_ * (node->type() == baldr::NodeType::kBorderControl);
-    c += gate_cost_ * (node->type() == baldr::NodeType::kGate);
+    c += gate_cost_ * (node->type() == baldr::NodeType::kGate) * (!InitiallyAllowed(node));
```

Comment:
why do you use `Allowed` method instead of simple `access_indication` flag ?
```suggestion
    c += gate_cost_ * (node->type() == baldr::NodeType::kGate) * (!node->access_indication());
```